File: flutter/lib/models/station.dart
class Station {
  final int id;
  final String nom;
  final String? adresse;
  final String? ville;
  final bool actif;

  Station({
    required this.id,
    required this.nom,
    this.adresse,
    this.ville,
    required this.actif,
  });

  factory Station.fromJson(Map<String, dynamic> json) {
    return Station(
      id: json['id'],
      nom: json['nom'],
      adresse: json['adresse'],
      ville: json['ville'],
      actif: json['actif'],
    );
  }
}

File: model/Station.java
package com.chams.tpe_backend.model;
import jakarta.persistence.*;
@Entity
@Table(name = "Station")
public class Station {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id ;

    @Column(nullable = false)
    private String nom ;
    private String adresse ;
    private String ville ;
    private boolean actif = true ;
    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getNom() {
        return nom;
    }

    public void setNom(String nom) {
        this.nom = nom;
    }

    public String getAdresse() {
        return adresse;
    }

    public void setAdresse(String adresse) {
        this.adresse = adresse;
    }

    public String getVille() {
        return ville;
    }

    public void setVille(String ville) {
        this.ville = ville;
    }

    public boolean isActif() {
        return actif;
    }

    public void setActif(boolean actif) {
        this.actif = actif;
    }
}
